packages/mobile/src/backup/BackupQuiz.tsx

```diff
@@ -111,28 +119,32 @@ export class BackupQuiz extends React.Component<Props, State> {
   onPressReset = () => {
     const mnemonic = this.getMnemonicFromNavProps()
     this.setState({
-      mnemonicWords: this.getShuffledWordSet(mnemonic),
+      mnemonicWords: getShuffledWordSet(mnemonic),
       userChosenWords: [],
+      is: Mode.entering,
     })
   }
 
-  onPressSubmit = () => {
+  afterCheck = () => {
     const { userChosenWords, mnemonicLength } = this.state
     const mnemonic = this.getMnemonicFromNavProps()
-    if (
-      userChosenWords.length === mnemonicLength &&
-      userChosenWords.map((w) => w.word).join(' ') === mnemonic
-    ) {
+    const lengthsMatch = userChosenWords.length === mnemonicLength
+
```

Comment:
nit: we can have here `const contentMatch = userChosenWords.map((w) => w.word).join(' ') === mnemonic` and later `if (lengthsMatch && contentMatch)`. Not a big deal, but would improve readability a bit.

Reply:
hmm if i do that then it will always do this mapping even if lengths tont match which seems wasteful

Reply:
I extracted out a function. best of both worlds!